graylog2-server/src/main/java/org/graylog/plugins/sidecar/migrations/V20180212165000_AddDefaultCollectors.java

```diff
@@ -569,22 +578,37 @@ private void ensureConfigurationVariable(String name, String description, String
         }
     }
 
-    private void ensureDefaultConfiguration(String name, Collector collector) {
+    private void ensureDefaultConfiguration(String name, CollectorState collectorState) {
         Configuration config = null;
         try {
             config = configurationService.findByName(name);
             if (config == null) {
                 LOG.debug("Couldn't find sidecar default configuration'{}' fixing it.", name);
                 throw new IllegalArgumentException();
+            } else {
+                if (collectorState.updatedTemplate && !config.template().equals(collectorState.collector.defaultTemplate())) {
```

Comment:
@mpfz0r We currently only maintain CRC for the collector `default_template` field. So if it was OK to update that field (`collectorState.updatedTemplate`) then we also update the corresponding data in the default config. 

This would only fail if you had changed your default config, but not updated your collector. Seems fairly unlikely. Do you think it's necessary to repeat the CRC check for the config template? Then we'd have to save the pre-check CRC, in case the collector `default_template` is overwritten and a new CRC generated. We could pass that value in `CollectorState`.

Reply:
> This would only fail if you had changed your default config, but not updated your collector. Seems fairly unlikely.

I think not. Actually this is the easiest way for the user to get things running.
And even if it's unlikely, we need to be sure.
So yeah, we need to check the checksum against the default config.
Can't we make a static version of  Collector#defaultTemplateUnchanged() that accepts a template as an argument?
We could use that one for the collector and the config then..

Reply:
Added the CRC-check to configurations. But with the current design we cannot properly support independent updates of collectors and configurations. 
We could
- add a separate CRC to configurations
- or better yet: revise the design so that default configurations and collectors are not user-editable.

Reply:
In hindsight, I think we should've never stored the checksum in the collector itself, but in a migration specific entry in the cluster collection.
We could keep a list of all known checksums there. it would be accessible for collectors and configs alike.

Reply:
@patrickmann  How about this instead?
https://github.com/Graylog2/graylog2-server/pull/17624